frontend/src/commit-panel/commit/Commit.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+
+import * as portal from '../../common/portal';
+import Buttons from './Buttons';
+import Form from './Form';
+
+interface CommitProps {
+  onCommit(message: string): void;
+  onDiscard(): void;
+}
+
+interface CommitState {
+  isFormVisible?: boolean;
+  commitMessage?: string;
+}
+
+export default class Commit extends React.Component<CommitProps, CommitState> {
+
+  state = {
+    isFormVisible: false,
+    commitMessage: '',
+  };
+
+  onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    this.props.onCommit(this.state.commitMessage);
+
+    this.setState({
+      commitMessage: '',
+    });
+  };
+
+  onCommitClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+
+    this.setState({
+      isFormVisible: true,
+    });
+  };
+
+  onCommitMessageChange = (e: React.FormEvent) => {
+    this.setState({
+      commitMessage: (e.target as any).value,
```

Comment:
`(e.target as HTMLTextAreaElement)`

Can be extracted directly in the attribute:
```
(`({ target: HTMLTextAreaElement })`)
```
 or in the body:
```
const target = e.target as HTMLTextAreaElement;
```